tfhe/src/core_crypto/algorithms/lwe_private_functional_packing_keyswitch.rs

```diff
@@ -286,3 +286,79 @@ pub fn private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphert
         slice_wrapping_add_assign(output.as_mut(), buffer.as_ref());
     }
 }
+
+/// Parallel variant of
+/// [`private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphertext`].
+///
+/// This will use all threads available in the current rayon thread pool.
+pub fn par_private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphertext<
+    Scalar,
+    KeyCont,
+    InputCont,
+    OutputCont,
+>(
+    lwe_pfpksk: &LwePrivateFunctionalPackingKeyswitchKey<KeyCont>,
+    output: &mut GlweCiphertext<OutputCont>,
+    input: &LweCiphertextList<InputCont>,
+) where
+    Scalar: UnsignedTorus,
+    KeyCont: Container<Element = Scalar>,
+    InputCont: Container<Element = Scalar>,
+    OutputCont: ContainerMut<Element = Scalar>,
+{
+    let thread_count = ThreadCount(rayon::current_num_threads());
+    par_private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphertext_with_thread_count(
+        lwe_pfpksk,
+        output,
+        input,
+        thread_count,
+    );
+}
+
+/// Parallel variant of
+/// [`private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphertext`].
+///
+/// This will try to use `thread_count` threads for the computation, if this number is bigger than
+/// the available number of threads in the current rayon thread pool then only the number of
+/// available threads will be used. Note that `thread_count` cannot be 0.
+pub fn par_private_functional_keyswitch_lwe_ciphertext_list_and_pack_in_glwe_ciphertext_with_thread_count<
+    Scalar,
+    KeyCont,
+    InputCont,
+    OutputCont,
+>(
+    lwe_pfpksk: &LwePrivateFunctionalPackingKeyswitchKey<KeyCont>,
+    output: &mut GlweCiphertext<OutputCont>,
+    input: &LweCiphertextList<InputCont>,
+    thread_count: ThreadCount,
+) where
+    Scalar: UnsignedTorus,
+    KeyCont: Container<Element = Scalar>,
+    InputCont: Container<Element = Scalar>,
+    OutputCont: ContainerMut<Element = Scalar>,
+{
+    assert!(input.lwe_ciphertext_count().0 <= output.polynomial_size().0);
```

Comment:
missing an assert on the thread count ? maybe it's missing in the other implem as well

Reply:
This function calls `par_private_functional_keyswitch_lwe_ciphertext_into_glwe_ciphertext_with_thread_count()` which contains the required `assert!`

Reply:
Fixed with implementation of suggestion below.